packages/flutter/lib/src/material/about.dart

```diff
@@ -384,6 +463,122 @@ class AboutDialog extends StatelessWidget {
   }
 }
 
+class _AdaptiveAboutDialog extends AboutDialog {
+  const _AdaptiveAboutDialog({
+    super.key,
+    super.applicationName,
+    super.applicationVersion,
+    super.applicationIcon,
+    super.applicationLegalese,
+    super.children,
+  });
+
+  List<Widget>? _actions(BuildContext context) {
+    final ThemeData themeData = Theme.of(context);
+    final MaterialLocalizations localizations =
+        MaterialLocalizations.of(context);
+    switch (themeData.platform) {
+      case TargetPlatform.iOS:
+      case TargetPlatform.macOS:
+        return <Widget>[
+          CupertinoDialogAction(
+            child: Text(themeData.useMaterial3
+                ? localizations.viewLicensesButtonLabel
+                : localizations.viewLicensesButtonLabel.toUpperCase()),
+            onPressed: () {
+              showLicensePage(
+                context: context,
+                applicationName: applicationName,
+                applicationVersion: applicationVersion,
+                applicationIcon: applicationIcon,
+                applicationLegalese: applicationLegalese,
+              );
+            },
+          ),
+          CupertinoDialogAction(
+            child: Text(themeData.useMaterial3
+                ? localizations.closeButtonLabel
+                : localizations.closeButtonLabel.toUpperCase()),
+            onPressed: () {
+              Navigator.pop(context);
+            },
+          ),
+        ];
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.linux:
+      case TargetPlatform.windows:
+        return <Widget>[
+          TextButton(
+            child: Text(themeData.useMaterial3
+                ? localizations.viewLicensesButtonLabel
+                : localizations.viewLicensesButtonLabel.toUpperCase()),
+            onPressed: () {
+              showLicensePage(
+                context: context,
+                applicationName: applicationName,
+                applicationVersion: applicationVersion,
+                applicationIcon: applicationIcon,
+                applicationLegalese: applicationLegalese,
+              );
+            },
+          ),
+          TextButton(
+            child: Text(themeData.useMaterial3
+                ? localizations.closeButtonLabel
+                : localizations.closeButtonLabel.toUpperCase()),
+            onPressed: () {
+              Navigator.pop(context);
+            },
+          ),
+        ];
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    super.build(context);
+
+    final String name = applicationName ?? _defaultApplicationName(context);
+    final String version =
+        applicationVersion ?? _defaultApplicationVersion(context);
```

Comment:
Prefer LHS and RHS in variable assignments on the same line as per the style guide.